### PR TITLE
Server 3.1 Docs Updates (DO NOT MERGE TO MASTER)

### PR DESCRIPTION
**Pull request by annabaker**
Updates in way for server 3.1.

TO DO:

- [x]  Update server architecture diagram  (owned by Andrew Song)
- [x] ~check over for consistency of "management console" over "kots ui"/"kots admin" etc~ too big for this ticket, making a separate card
- [x] check lowercase 'server'
- [x] update PDFs

### CODE REVIEW

**Comment by rosieyohannan** [comment]
To Do:
- [ ] Remove coming soon notes https://circleci.atlassian.net/browse/SERVER-1140

File: jekyll/_cci2/server-3-overview.adoc [hunk 1]
```diff
@@ -14,11 +14,11 @@ toc::[]
...
 == Introduction
 
-CircleCI Server v{serverversion} is a continuous integration and continuous delivery (CI/CD) platform that you can install on your GCP or AWS Kubernetes cluster.
+CircleCI server v{serverversion} is a continuous integration and continuous delivery (CI/CD) platform that you can install on your GCP or AWS Kubernetes cluster.
 
-The core of the CircleCI Server application runs inside Kubernetes.
+The core of the CircleCI server application runs inside Kubernetes.
 
-The application exposes five services using load balancers, three of these load balancers are VPC-internal.
+The application exposes 4 services using load balancers, three of these load balancers are VPC-internal.
```
Comment (rosieyohannan) [inline jekyll/_cci2/server-3-overview.adoc:21]: Add a note to say the external load balancer can be made private/internal now with link to more info https://circleci.atlassian.net/browse/SERVER-1156

File: jekyll/_cci2/server-3-install.adoc [hunk 2]
```diff
@@ -143,17 +143,84 @@ The domain you selected for your CircleCI installation is the Homepage URL and *
...
-. *AWS S3* (required): Region your S3 bucket is in.
-. *AWS IAM Access Key ID* (required): https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html[AWS Access Key ID] for S3 bucket access.
-. *AWS IAM Secret Key* (required): https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html[IAM Secret Key] for S3 bucket access.
+Server 3.x hosts build artifacts, test results, and other state in object storage. We support
+
+. Any https://docs.aws.amazon.com/s3/index.html?nc2=h_ql_doc_s3[S3]-compatible storage
+. https://cloud.google.com/storage/[Google Cloud Storage]
+
+While any S3-compatible object storage may work, we test and support https://aws.amazon.com/s3/[AWS S3] and https://min.io[Minio]. For object storage
+providers that do not support the https://docs.aws.amazon.com/AmazonS3/latest/API/Type_API_Reference.html[S3 API], such as
```
Comment (rosieyohannan) [inline jekyll/_cci2/server-3-install.adoc:202]: @nsmith5 Hey, here we say we support any S3 storage but then say we suport AWS S3 and Minio, but others may work - so I guess not officially supported. My question is would it be better for the bullet list  to say AWS S3, Minio and GCS are supported?

**Comment by rosieyohannan** [comment]
To do:

- [ ] there are a bunch of links to server docs from elsewhere, lots in the config ref, these should all probably point to 3.x server docs now.
- [ ] add upgrade guide

File: jekyll/_cci2/server-3-install.adoc (hunk 2 above)
Comment (nsmith5) [reply jekyll/_cci2/server-3-install.adoc:202]: Right, that is definitely contradictory. I think the list saying AWS S3, Minio and GCS are supported makes sense like you say

File: jekyll/_cci2/server-3-install.adoc [hunk 3]
```diff
@@ -143,17 +143,84 @@ The domain you selected for your CircleCI installation is the Homepage URL and *
...
-A *Storage Bucket Name* is required, in addition to the following fields, depending on if you are using AWS or GCP. Ensure
-that the bucket name you provide exists in your chosen object storage provider before proceeeding.
 
-#### AWS S3
-. *AWS S3* (required): Region your S3 bucket is in.
-. *AWS IAM Access Key ID* (required): https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html[AWS Access Key ID] for S3 bucket access.
-. *AWS IAM Secret Key* (required): https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html[IAM Secret Key] for S3 bucket access.
+Server 3.x hosts build artifacts, test results, and other state in object storage. We support
+
+. Any https://docs.aws.amazon.com/s3/index.html?nc2=h_ql_doc_s3[S3]-compatible storage
```
Comment (rosieyohannan) [inline jekyll/_cci2/server-3-install.adoc:198]:
```suggestion
. https://aws.amazon.com/s3/[AWS S3]
. https://min.io[Minio]
```

File: jekyll/_cci2/server-3-overview.adoc (hunk 1 above)
Comment (rosieyohannan) [reply jekyll/_cci2/server-3-overview.adoc:21]: @nathanfishcircleci Just checking, is this the right place for a link to a new "Load Balancers" section, as per the jira ticket mentioned?

**Comment by rosieyohannan** [comment]
To do:
- [ ] check over for consistency of "management console" over "kots ui"/"kots admin" etc

File: jekyll/_cci2/server-3-install-migration.adoc [hunk 4]
```diff
@@ -10,64 +10,67 @@ version:
...
-Migrating from 2.19.x to 3.0 requires that you backup your 2.19 installation and then restore the data to the new Server 3.0 installation.
-It does require an already operating Server 3.0 installation. Depending on the size of your data stores, the migration
+Migrating from 2.19.x to 3.x requires that you backup your 2.19.x installation and then restore the data to the new server 3.x installation.
+It does require an already operating server 3.0 installation. Depending on the size of your data stores, the migration
 can take anywhere from a few minutes to a few hours. We recommend using a staging environment before completing this
 process in a production environment. This will not only allow you to gain a better understanding of the migration process,
 but will also give you a feel for how long the migration will take to complete.
 
-NOTE: If you have externalized your data stores (Mongo, Postgres, and Vault), then you need to wait until the Server 3.1
+NOTE: If you have externalized your data stores (Mongo, Postgres, and Vault), then you need to wait until the server 3.1
```
Comment (rosieyohannan) [inline jekyll/_cci2/server-3-install-migration.adoc:19]:
```suggestion
```

File: jekyll/_cci2/server-3-install.adoc [hunk 5]
```diff
@@ -143,17 +177,85 @@ The domain you selected for your CircleCI installation is the Homepage URL and *
...
+EOF
+
+vault token create -policy=circleci
+----
+
+Create a token using the policy:
+[source,sh]
+----
+vault token create -policy=circleci -period=30m
+----
```
Comment (rosieyohannan) [inline jekyll/_cci2/server-3-install.adoc:226]:
```suggestion
. Token: The Vault token to be used by CircleCI. 
```

File: jekyll/_cci2/server-3-install.adoc [hunk 6]
```diff
@@ -143,17 +177,85 @@ The domain you selected for your CircleCI installation is the Homepage URL and *
...
+
```
[the remark this answers is not in the document]
Comment (rosieyohannan) [reply jekyll/_cci2/server-3-install.adoc]: removed for now :-)

**Review by sarameisburger — APPROVED** [review]

**Pull request merged**